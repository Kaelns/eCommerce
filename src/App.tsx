import { ThemeProvider } from '@emotion/react';
import { Router } from '@/features/Router/Router';
import { lightTheme } from '@/data/theme/lightTheme';
import { AuthContextProvider } from '@/context/AuthContext/AuthContext';

export function App(): JSX.Element {
  return (
    <ThemeProvider theme={lightTheme}>
      <AuthContextProvider>
        <Router />
      </AuthContextProvider>
    </ThemeProvider>
  );
}
