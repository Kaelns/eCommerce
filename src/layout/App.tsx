import type { SxStyles } from '@/shared/types/types';
import { Stack } from '@mui/system';
import { Alert } from '@/features/alert';
import { Header } from '@/layout/Header';
import { Outlet } from 'react-router-dom';
import { SectionContainer } from '@/layout/SectionContainer';
import { useStartSessionQuery } from '@/services/ecommerce-api';
import { SuspenseWithError } from '@/components/SuspenseWithError';
import { PageSkeleton } from '@/components/skeleton/PageSkeleton';

const sxStyles: SxStyles = {
  container: {
    flex: 1
  },

  sectionContainer: {
    py: 3.5,
    px: { zero: 1, tablet: 2 },
    gap: 1.5
  }
};

export function App(): React.ReactNode {
  const { /* data, */ isLoading, isError /* , error */ } = useStartSessionQuery();

  return (
    <>
      <Header />

      <Stack component="main" gap={1.5} sx={sxStyles.container}>
        <SectionContainer sx={sxStyles.sectionContainer}>
          <SuspenseWithError settings={{ isLoading, isError /* , error: error?.data?.message */ }} components={{ Skeleton: <PageSkeleton /> }}>
            <Outlet />
          </SuspenseWithError>
        </SectionContainer>
        <Alert />
      </Stack>
    </>
  );
}
