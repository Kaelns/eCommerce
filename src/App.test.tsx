import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/vitest';
import App from './App';

test('renders learn react link', () => {
  render(<App />);
  const linkElement = screen.getByText(/Welcome to eCommerce Application/i);
  expect(linkElement).toBeInTheDocument();
});
