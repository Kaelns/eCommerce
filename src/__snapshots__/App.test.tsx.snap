// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Given App component > When rendered, should match snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation2 MuiAppBar-root MuiAppBar-colorDefault MuiAppBar-positionStatic css-zwpp1z-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg _sectionContainer_b162ba _headerContainer_7110fc css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="_columnContainer_7110fc MuiBox-root css-0"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeLarge css-kxf97k-MuiSvgIcon-root"
          data-testid="TollIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15 4c-4.42 0-8 3.58-8 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8m0 14c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6M3 12c0-2.61 1.67-4.83 4-5.65V4.26C3.55 5.15 1 8.27 1 12s2.55 6.85 6 7.74v-2.09c-2.33-.82-4-3.04-4-5.65"
          />
        </svg>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-9c2k02-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiBadge-root css-1c32n2y-MuiBadge-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-19zg5yn-MuiSvgIcon-root"
              data-testid="ShoppingCartOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.55 13c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.37-.66-.11-1.48-.87-1.48H5.21l-.94-2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2zM6.16 6h12.15l-2.76 5H8.53zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2m10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2"
              />
            </svg>
            <span
              class="MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorPrimary css-1fx6c1z-MuiBadge-badge"
            >
              0
            </span>
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-9c2k02-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiBadge-root _badge_2e3992 css-1c32n2y-MuiBadge-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-19zg5yn-MuiSvgIcon-root"
              data-testid="AccountCircleOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2M7.35 18.5C8.66 17.56 10.26 17 12 17s3.34.56 4.65 1.5c-1.31.94-2.91 1.5-4.65 1.5s-3.34-.56-4.65-1.5m10.79-1.38C16.45 15.8 14.32 15 12 15s-4.45.8-6.14 2.12C4.7 15.73 4 13.95 4 12c0-4.42 3.58-8 8-8s8 3.58 8 8c0 1.95-.7 3.73-1.86 5.12"
              />
              <path
                d="M12 6c-1.93 0-3.5 1.57-3.5 3.5S10.07 13 12 13s3.5-1.57 3.5-3.5S13.93 6 12 6m0 5c-.83 0-1.5-.67-1.5-1.5S11.17 8 12 8s1.5.67 1.5 1.5S12.83 11 12 11"
              />
            </svg>
            <span
              class="MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorPrimary _badgeIcon_2e3992 _badgeIcon_2e3992 css-1i43is6-MuiBadge-badge"
            >
              Login
            </span>
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-9c2k02-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-19zg5yn-MuiSvgIcon-root"
            data-testid="MenuIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
  <main
    class="_container_2da750 MuiBox-root css-0"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg _sectionContainer_b162ba _sectionContainer_2da750 css-1oqqzyl-MuiContainer-root"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-1mx4yc8-MuiTypography-root"
      >
        Main page
      </h5>
      <nav
        class="MuiBox-root css-0"
      >
        <div
          class="MuiTabs-root MuiTabs-vertical css-1g4ac1o-MuiTabs-root"
        >
          <div
            class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
            style="overflow: hidden; margin-right: 0px;"
          >
            <div
              aria-orientation="vertical"
              class="MuiTabs-flexContainer MuiTabs-flexContainerVertical css-lfwcke-MuiTabs-flexContainer"
              role="tablist"
            >
              <button
                aria-selected="true"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected _buttons_f95cce _horizontalButton_f95cce css-mtjpl7-MuiButtonBase-root-MuiTab-root"
                role="tab"
                tabindex="0"
                type="button"
              >
                Main
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary _buttons_f95cce _horizontalButton_f95cce css-mtjpl7-MuiButtonBase-root-MuiTab-root"
                role="tab"
                tabindex="-1"
                type="button"
              >
                User
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary _buttons_f95cce _horizontalButton_f95cce css-mtjpl7-MuiButtonBase-root-MuiTab-root"
                role="tab"
                tabindex="-1"
                type="button"
              >
                Error
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary _buttons_f95cce _horizontalButton_f95cce css-mtjpl7-MuiButtonBase-root-MuiTab-root"
                role="tab"
                tabindex="-1"
                type="button"
              >
                Login
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary _buttons_f95cce _horizontalButton_f95cce css-mtjpl7-MuiButtonBase-root-MuiTab-root"
                role="tab"
                tabindex="-1"
                type="button"
              >
                Basket
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary _buttons_f95cce _horizontalButton_f95cce css-mtjpl7-MuiButtonBase-root-MuiTab-root"
                role="tab"
                tabindex="-1"
                type="button"
              >
                Catalog
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary _buttons_f95cce _horizontalButton_f95cce css-mtjpl7-MuiButtonBase-root-MuiTab-root"
                role="tab"
                tabindex="-1"
                type="button"
              >
                About Us
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary _buttons_f95cce _horizontalButton_f95cce css-mtjpl7-MuiButtonBase-root-MuiTab-root"
                role="tab"
                tabindex="-1"
                type="button"
              >
                Register
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary _buttons_f95cce _horizontalButton_f95cce css-mtjpl7-MuiButtonBase-root-MuiTab-root"
                role="tab"
                tabindex="-1"
                type="button"
              >
                Detailed Product
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <span
              class="MuiTabs-indicator css-13dyh3w-MuiTabs-indicator"
              style="top: 0px; height: 0px;"
            />
          </div>
        </div>
      </nav>
    </div>
  </main>
</DocumentFragment>
`;
